include/matrix_operations/EigenvaluesProblems.h

```diff
@@ -424,6 +424,54 @@ class EigenvaluesProblems<NT, Eigen::Matrix<NT,Eigen::Dynamic,Eigen::Dynamic>, E
 
         return lambdaMinPositive;
     }
+
+    // Using LDLT decomposition to check membership
+    // Faster than computing the largest eigenvalue with Spectra 
+    // more numerically stable for singular matrices
+    bool isPositiveSemidefinite(MT const &A) {
+        Eigen::LDLT<MT> A_ldlt(A);
+        if (A_ldlt.info() != Eigen::NumericalIssue && A_ldlt.isPositive())
+            return true;
+        return false;
+    }
+
+    /// Minimum positive eigenvalue of the generalized eigenvalue problem A - lB
+    /// Use Eigen::GeneralizedSelfAdjointEigenSolver<MT> ges(B,A) (faster)
+    /// \param[in] A: symmetric positive definite matrix
+    /// \param[in] B: symmetric matrix
+    /// \return The minimum positive eigenvalue and the corresponding eigenvector
+    NT minPosLinearEigenvalue_EigenSymSolver(MT const & A, MT const & B, VT &eigvec) {
+        NT lambdaMinPositive = NT(0);
+#if defined(SPECTRA_SYM_EIGENVALUES_SOLVER)
+//         int matrixDim = A.rows();
+//         int ncv = 15 < matrixDim ? 15 : matrixDim; // convergence speed
+
+//         Spectra::DenseSymMatProd<NT> op(B);
```

Comment:
Please delete uncommented code that you don't need

Reply:
I removed this.